adventure/adventure.py

```diff
@@ -2936,15 +2938,32 @@ def E(t: str) -> str:
                 continue
             bonus_tinkerer = c.att + c.skill["att"]
             if c.heroclass["name"] == "Tinkerer" and not sharpen:
-                sharpen_chance = min(int(bonus_tinkerer / 2.5 + 1), 25)
+                sharpen_chance = min(int(bonus_tinkerer / 2.5 + 1), c.lvl)
                 sharpen_roll = random.randint(1, 100)
                 if sharpen_roll in range (1, sharpen_chance):
                     sharpen = True
-                    sharpen_bonus = max(int(sharpen_chance / 10 * 4), 2)
-                    msg += f"{bold(self.E(user.display_name))} sharpened the weapons of the party! 🗡+{sharpen_bonus}%.\n"     
+                    sharpen_bonus = int(sharpen_chance * 0.4) + 3
+                    msg += f"{bold(self.E(user.display_name))} sharpened the weapons of the party! 🗡+{sharpen_bonus}%.\n"
+        for user in session.pray: #check if a cleric is praying and calculate the possible bonus
+            try:
+                c = await Character._from_json(self.config, user)
+            except Exception:
+                log.error("Error with the new character sheet", exc_info=True)
+                continue
+            bonus_cleric = int((c.int + c.skill["int"] + c.att + c.skill["att"] + c.cha + c.skill["cha"])/3)
+            if c.heroclass["name"] == "Cleric" and not aura:
+                aura_chance = min(int(bonus_cleric / 2 + 1), c.lvl)
+                aura_roll = random.randint(1, 100)
+                if aura_roll in range (1, aura_chance):
+                    aura = True
+                    aura_bonus = int(aura_chance * 0.2)
+                    msg += (
+                        f"An holy aura starts surrounding {bold(self.E(user.display_name))} while praying! "

```

Comment:
```suggestion
                        f"A holy aura starts surrounding {bold(self.E(user.display_name))} while praying! "
```
Confusingly... "an" can be used before some words that start with "h" like "an hour" but normally when the "h" sound isn't really pronounced... In the case of "holy" you pronounce the "h" so it's best to say "a holy" not "an holy"